arbitrum-docs/devs-how-tos/how-to-estimate-gas.mdx

```diff
@@ -0,0 +1,142 @@
+
+- P (L2 Gas Price) ⇒ Price to pay for each gas unit. It starts at 0.1 gwei on Arbitrum One (0.01 gwei on Arbitrum Nova) and can increase depending on the demand for network resources.
+    - Call `ArbGasInfo.getPricesInWei()` and get the sixth element
+- L2G (Gas used on L2) ⇒ Gas used to compute the transaction on L2. This does not include the “*posting on L1”* part of the calculations. The value of L2G will depend on the transaction itself, but having the data of the transaction, we can calculate it as follows:
+    - Call `NodeInterface.GasEstimateComponents()` with the transaction data and subtract the second element (`gasEstimateForL1`, which estimates the L1 part of the fees) from the first (`gasEstimate`, which includes both the L1 and the L2 parts).
+- L1P (L1 estimated price per byte of data) ⇒ Estimated cost of posting 1 byte of data on L1:
+    - Call `ArbGasInfo.getPricesInWei()` and get the second element
+- L1S (Size of data to be posted on L1, in bytes) ⇒ This will depend on the data of the transaction. Keep in mind that Arbitrum adds a fixed amount to this number to make up for the static part of the transaction, which is also posted on L1 (140 bytes).
+    - For Arbitrum Nova (AnyTrust), the size of the data is also a fixed value, as only the Data Availability Certificate is posted on L1, [as explained here](/inside-anytrust#data-availability-certificates).
+
+## An example of how to apply this formula in your code
+
+Finally, we show an example of how to get the values we just described and how to estimate the gas usage of a transaction in Javascript. We will use our [SDK](https://github.com/OffchainLabs/arbitrum-sdk) to connect to `ArbGasInfo` and `NodeInterface`.
+
+We first instantiate a factory object for ArbGasInfo and NodeInterface, using two methods from the SDK. `l2Provider` is a regular JSON RPC provider for the L2 network we are using, and both `ARB_GAS_INFO` and `NODE_INTERFACE_ADDRESS` are the addresses of the precompiles in said network.
+
+```ts
+const { ArbGasInfo__factory } = require("@arbitrum/sdk/dist/lib/abi/factories/ArbGasInfo__factory");
+const { NodeInterface__factory } = require("@arbitrum/sdk/dist/lib/abi/factories/NodeInterface__factory");
+const { ARB_GAS_INFO, NODE_INTERFACE_ADDRESS } = require("@arbitrum/sdk/dist/lib/dataEntities/constants");
+
+...
+
+// Instantiation the ArbGasInfo and NodeInterface objects
+const arbGasInfo = ArbGasInfo__factory.connect(
+    ARB_GAS_INFO,
+    l2Provider
+);
+const nodeInterface = NodeInterface__factory.connect(
+    NODE_INTERFACE_ADDRESS,
+    l2Provider
+);
+```
+
+For this example, we will use the method `ArbGasInfo.getPricesInWei()` and `NodeInterface.gasEstimateComponents()` to get all the information we need. For the gasEstimateComponents() call we will pass a destinationAddress (might be any address) and the data we want to send, to get results as accurate as possible.
```

Comment:
```suggestion
For this example, we'll use the method `ArbGasInfo.getPricesInWei()` and `NodeInterface.gasEstimateComponents()` to get the information we need. For the `gasEstimateComponents()` call, we'll pass a `destinationAddress` (this can be any address) and the data we want to send.
```

meta -> 

 - conversational tone can help keep readers in flow; contractions can help with conversational tone
 - consistent application of ticks for code formatting
 - cutting some copy that doesn't add value (accuracy is implied, but let's riff on this if I'm misunderstanding the intent)